cli/src/commands/branch.rs

```diff
@@ -399,6 +393,13 @@ fn cmd_branch_set(
     Ok(())
 }
 
+/// Whether or not the `branch` has any tracked remotes (i.e. is a tracking
+/// local branch.)
+fn has_tracked_remote_branches(view: &View, branch: &str) -> bool {
```

Comment:
(Optional) Re making the code more difficult from https://github.com/martinvonz/jj/pull/3884/commits/88ff3555ff40e8e70e51fdd5932ce97cbdadbcdd#r1640243864: I don't know that it would be that much more difficult to return an iterator or a vector here, but maybe I'm missing something.

In any case, adding the remote names to the message (or not) is non-blocking for the PR.

Reply:
It can return a `Vec`. Returning an iterator has lifetime issue.

That said, the things I wasn't bothered to are to provide different hints depending on the remote name (`@git` vs the others), and writing tests for various combinations.